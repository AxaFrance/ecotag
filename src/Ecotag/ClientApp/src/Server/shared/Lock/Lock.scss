.af-btn {
  width: 25%
}

.lock {
  &__button-container {
    display: flex;
    justify-content: center;
  }

  &__button {
    &--disabled {
      display: none;
    }
  }

  &__button-icon {
    margin-left: 5%;
  }
}