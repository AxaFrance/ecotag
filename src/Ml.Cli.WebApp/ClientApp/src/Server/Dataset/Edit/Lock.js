import React from "react";
import Button from '@axa-fr/react-toolkit-button';

const Lock = ({state, setState}) => {
    const buttonDisabled = state.filesSend.length === 0;
    const lockDataset = () => {
        setState({...state, openLockModal: !state.openLockModal});
    };

    return (
        <div className="edit-dataset__lock-button-container">
            <Button
                id="lock-button"
                classModifier={buttonDisabled || state.isLock ? 'disabled' : 'danger'}
                onClick={lockDataset}
                icon="lock"
                disabled={buttonDisabled}>
                <span className="af-btn__text">{state.isLock ? 'Dataset verrouillée' : "verrouillée"}</span>
                <i className=' edit-dataset__lock-button-icon glyphicon glyphicon-lock' />
            </Button>
        </div>
    );
};

export default Lock;
