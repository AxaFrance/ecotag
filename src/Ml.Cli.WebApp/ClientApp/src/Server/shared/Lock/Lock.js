import React from "react";
import Button from '@axa-fr/react-toolkit-button';
import './Lock.scss';

const Lock = ({isLocked, isDisabled, text, lockedText, onLockAction}) => {
    return (
        <div className="lock__button-container">
            <Button
                id="lock-button"
                classModifier={isDisabled || isLocked ? 'disabled' : 'danger'}
                onClick={onLockAction}
                icon="lock"
                disabled={isDisabled}>
                <span className="af-btn__text">{isLocked ? lockedText : text}</span>
                <i className='lock__button-icon glyphicon glyphicon-lock' />
            </Button>
        </div>
    );
};

export default Lock;
