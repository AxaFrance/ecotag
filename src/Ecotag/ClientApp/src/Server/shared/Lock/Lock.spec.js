import {render} from '@testing-library/react';
import Lock from "./Lock";

describe('Lock', () => {
    it('should render correctly', () => {
        const {asFragment} = render(<Lock isLocked={false} isDisabled={false} onLockAction={() => {
        }} text="Unlocked text" lockedText="Locked text"/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
