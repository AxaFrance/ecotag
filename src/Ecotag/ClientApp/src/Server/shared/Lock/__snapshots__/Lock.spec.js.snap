// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lock should render correctly 1`] = `
<DocumentFragment>
  <div
    class="lock__button-container"
  >
    <button
      class="btn af-btn af-btn--danger"
      icon="lock"
      id="lock-button"
    >
      <span
        class="af-btn__text"
      >
        Unlocked text
      </span>
      <i
        class="lock__button-icon glyphicon glyphicon-lock"
      />
    </button>
  </div>
</DocumentFragment>
`;
